### DISCUSSION TITLE
Help required regarding converting a legacy Silverlight-CSLA.NET application

### DISCUSSION THREAD

**Discussion by vikaskumar82** [MarimerLLC/csla#1781] [Questions]
Hello Admin,
We have developed a silverlight application using CSLA.NET and we were trying to maintain it for years. But now it is a dead end for silverlight. The product is taxation related and has a support for last 9-10 years to leverage the legislation rules/forms coming through every year. But we cannot change to Angular or HTML in couple of months even. 
What is the best and efficient solution to change it?
Plz suggest

**Comment by danielmartind** [comment]
Might be worth looking at OpenSilver. [ https://opensilver.net/]( https://opensilver.net/)

**Reply by BlagoCuljak** [reply]
Try Blazor, you will need to build interface, with HTML and CSS, but all logic behind is CSLA rules.

We had WPF app(XAML as Silverlight), and upgraded to CSLA 5.3 and now we have same logic for the program, just adding more and more interface (buttons, inputs...) to Blazor side, binding is almost the same.

**Comment by vikaskumar82** [comment]
Thank you for that. We already had a thought about blazor but problem is blazor doesn’t work in IE. However, since far, clients were instructed to use IE so that silver light can work well. Now if we convert into blazor, we are thinking to convert a few areas first. In that case, it is a confusion for clients that for some areas they are bound to use IE only and for some pages, they are restricted to use IE 
Regards,
Vikas Kumar Developer | Mentor | TrainerPh: +61 477646537LinkedIn: https://www.linkedin.com/in/vikas-kumar-3a7338117 

    On Tuesday, 25 August 2020, 04:47:18 AM GMT+10, BlagoCuljak <notifications@github.com> wrote:  
 
 


Try Blazor, you will need to build interface, with HTML and CSS, but all logic behind is CSLA rules.

We had WPF app(XAML as Silverlight), and upgraded to CSLA 5.3 and now we have same logic for the program, just adding more and more interface (buttons, inputs...) to Blazor side, binding is almost the same.

**Reply by rockfordlhotka** [reply]
I think you are going to run into challenges with IE in any case.

Generally speaking, all modern technologies won't work with IE - that's true of Blazor, but also modern Angular, OpenSilver, etc.

Your strategy almost has to revolve around building a new UI that uses modern evergreen browsers, and to migrate your users off IE/Silverlight and into the new app on modern browsers.

**Reply by BlagoCuljak** [reply]
If Microsoft is dropping IE altogther, you could also:
https://www.theverge.com/2020/8/17/21372487/microsoft-internet-explorer-11-support-end-365-legacy-edge

Some effort will go into this, but long-term, it's a wise decision.

**Reply by rockfordlhotka** [reply]
They most certainly ARE dropping IE altogether - they are just doing it slowly to allow time for people who are stuck on ancient technology to migrate forward. But the clock is surely ticking.

**Comment by vikaskumar82** [comment]
Thanks. Tried that too but it is too time consuming as some CSLA classes are not supported.
Regards,
Vikas Kumar Developer | Mentor | TrainerPh: +61 477646537LinkedIn: https://www.linkedin.com/in/vikas-kumar-3a7338117 

    On Wednesday, 26 August, 2020, 01:40:38 pm GMT+10, Rockford Lhotka <notifications@github.com> wrote:  
 
 


They most certainly ARE dropping IE altogether - they are just doing it slowly to allow time for people who are stuck on ancient technology to migrate forward. But the clock is surely ticking.
